src/services/expenses.service.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+let expenses = [];
+
+const resetExpenses = () => {
+  expenses = [];
+};
+
+const getExpenses = (userId, from, to, categories) => {
+  let expensesToDisplay = [...expenses];
+
+  if (userId) {
+    expensesToDisplay = expensesToDisplay.filter(e => e.userId === +userId);
+  }
+
+  if (categories) {
+    expensesToDisplay = expensesToDisplay
+      .filter(e => e.category === categories);
+  }
+
+  if (from) {
+    expensesToDisplay = expensesToDisplay.filter(e => (
+      Date.parse(e.spentAt) > Date.parse(from)
+    ));
+  }
+
+  if (to) {
+    expensesToDisplay = expensesToDisplay.filter(e => (
+      Date.parse(e.spentAt) < Date.parse(to)
+    ));
+  }
```

Comment:
please use condition in one `.filter` method, because now it will invoke filter a lot of times of there is a multiple filters in query